src/main/java/it/pagopa/pn/deliverypush/action/checkattachmentretention/CheckAttachmentRetentionHandler.java

```diff
@@ -0,0 +1,60 @@
+package it.pagopa.pn.deliverypush.action.checkattachmentretention;
+
+import it.pagopa.pn.deliverypush.action.startworkflow.notificationvalidation.AttachmentUtils;
+import it.pagopa.pn.deliverypush.config.PnDeliveryPushConfigs;
+import it.pagopa.pn.deliverypush.dto.ext.delivery.notification.NotificationInt;
+import it.pagopa.pn.deliverypush.dto.timeline.details.TimelineElementCategoryInt;
+import it.pagopa.pn.deliverypush.middleware.queue.producer.abstractions.actionspool.ActionType;
+import it.pagopa.pn.deliverypush.service.NotificationService;
+import it.pagopa.pn.deliverypush.service.SchedulerService;
+import it.pagopa.pn.deliverypush.service.TimelineService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.time.Instant;
+
+@Component
+@AllArgsConstructor
+@Slf4j
+public class CheckAttachmentRetentionHandler {
+    private final NotificationService notificationService;
+    private final TimelineService timelineService;
+    private final AttachmentUtils attachmentUtils;
+    private final PnDeliveryPushConfigs configs;
+    private final SchedulerService schedulerService;
+    
+    public void handleCheckAttachmentRetentionBeforeExpiration(String iun){
+        log.debug("Start handleCheckAttachmentRetentionBeforeExpiration - iun={}", iun);
+        boolean isNotificationRefined = timelineService.getTimeline(iun, false)
+                .stream().anyMatch(elem -> TimelineElementCategoryInt.REFINEMENT.equals(elem.getCategory()) || TimelineElementCategoryInt.NOTIFICATION_VIEWED.equals(elem.getCategory()));
+        
+        if(! isNotificationRefined){
```

Comment:
ha senso testare anche il cancelled?
inoltre, non è meglio sfruttare i metodi di TimelineUtils che così è più facilmente "ricercabile" dove si stanno eseguendo i check?

Potrebbe aver senso prevedere un metodo che faccia un check unico in base a refinement, viewed, canceled, ecc ecc (tramite batchgetitem).

Infine, nel caso di multidestinatario, non so quanto cercare genericamente che sia refinement/viewed abbia senso. potrei aver raggiungo il destinatario A subito e quindi dopo 10gg  scattano i 120. ma se il destinatario B "è lento", finisce che poi non ha più gli allegati disponibili